src/Microsoft.AspNetCore.Server.IntegrationTesting/TestMatrix.cs

```diff
@@ -0,0 +1,266 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting
+{
+    public class TestMatrix
+    {
+        public IList<ServerType> Servers { get; set; } = new List<ServerType>();
+        public IList<string> Tfms { get; set; } = new List<string>(); // Can derive from running TFM
+        public IList<ApplicationType> ApplicationTypes { get; set; } = new List<ApplicationType>(); // net461/CLR is always portable, derive from TFM
+        public IList<RuntimeArchitecture> Architectures { get; set; } = new List<RuntimeArchitecture>();
+
+        // ANCM specific...
+        public IList<HostingModel> HostingModels { get; set; } = new List<HostingModel>();
```

Comment:
Subclass for ANCM specific stuff?

Reply:
I'll look closer at that seperation as part of https://github.com/aspnet/Hosting/issues/1371